crates/types/src/invocation.rs

```diff
@@ -273,19 +273,95 @@ impl fmt::Display for InvocationTarget {
     }
 }
 
+/// Concurrency guarantee of the invocation request.
+#[derive(Debug, Default, Clone, PartialEq, Eq, serde::Serialize, serde::Deserialize)]
+enum ConcurrencyGuarantee {
+    /// Enqueue the invocation in the inbox when the target is busy
+    EnqueueWhenBusy {
+        /// fka ServiceId.name
+        inbox_target: ByteString,
+        /// fka ServiceId.key
+        inbox_key: ByteString,
+    },
+    /// No queueing, just execute the request
+    MaxParallelism,
+    /// Use the default from the target semantics
+    #[default]
+    UseTargetDefault,
```

Comment:
It's unclear what the default means here. It might be worth adding a couple of words to explain that it'll use the default behavior based on the target type (virtual object, service, etc.)

Also, see my note below.